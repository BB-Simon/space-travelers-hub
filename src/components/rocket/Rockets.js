import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets, reserveRocket } from '../../redux/features/rockets/rocketsSlice';

const Rockets = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.rockets);
  const status = useSelector((state) => state.rockets.status);

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  const handleReserveRocket = (id) => {
    dispatch(reserveRocket(id));
  };

  return (
    <div>
      {status === 'loading' && <div>Loading...</div>}
      {status === 'success' && (
      <>
        <div className="rockets-container">
          {rockets.map((rocket) => (
            <div key={rocket.id} className="rockets">
              {rocket.flickr_images.length > 0 && (
              <img src={rocket.flickr_images[0]} alt={rocket.name} />
              )}
              <div className="rockets-description">
                <h2>{rocket.name}</h2>
                <p className="rocket-para">{rocket.description}</p>
                {rocket.reserved ? (
                  <>
                    <span className="reserved-badge">Reserved</span>
                    <button
                      type="button"
                      className="cancel-reservation-btn"
                      onClick={() => handleReserveRocket(rocket.id)}
                    >
                      Cancel reservation
                    </button>
                  </>
                ) : (
                  <button
                    type="button"
                    className="reserve-rocket-btn"
                    onClick={() => handleReserveRocket(rocket.id)}
                  >
                    Reserve Rocket
                  </button>
                )}
              </div>
            </div>
          ))}
        </div>
      </>
      )}
      {status === 'rejected' && <div>Failed to load rockets data</div>}
    </div>
  );
};

export default Rockets;
