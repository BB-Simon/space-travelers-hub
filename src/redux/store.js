import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionsSlice';
import rocketsReducer from './features/rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
