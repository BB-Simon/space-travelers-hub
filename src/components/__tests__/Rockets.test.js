import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import Rockets from '../rocket/Rockets';
import store from '../../redux/store';

describe('Jest Snapshot testing suite', () => {
  it('renders correctly', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    const ele = screen.getByText(/Rockets data loading...!/i);
    expect(ele).toBeInTheDocument();
  });
});
