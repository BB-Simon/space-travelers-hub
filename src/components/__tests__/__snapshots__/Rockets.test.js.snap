// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite renders correctly 1`] = `
  <p>Rockets data loading...!</p>
`;